front/pages/w/[wId]/members/index.tsx

```diff
@@ -931,3 +928,68 @@ function ChangeMemberModal({
 function displayRole(role: RoleType): string {
   return role === "user" ? "member" : role;
 }
+
+function RoleDropDown({
+  selectedRole,
+  onChange,
+}: {
+  selectedRole: ActiveRoleType;
+  onChange: (role: ActiveRoleType) => void;
+}) {
+  return (
+    <DropdownMenu>
+      <DropdownMenu.Button>
+        <div className="group flex cursor-pointer items-center gap-2">
+          <Chip
+            color={ROLES_DATA[selectedRole]["color"]}
+            className="capitalize"
+          >
+            {displayRole(selectedRole)}
+          </Chip>
+          <IconButton
+            icon={ChevronDownIcon}
+            size="sm"
+            variant="secondary"
+            className="group-hover:text-action-400"
+          />
+        </div>
+      </DropdownMenu.Button>
+      <DropdownMenu.Items origin="topLeft">
+        {ACTIVE_ROLES.map((role) => (
+          <DropdownMenu.Item
+            key={role}
+            onClick={() => onChange(role)}
+            label={
+              displayRole(role).charAt(0).toUpperCase() +
+              displayRole(role).slice(1)
+            }
+          />
+        ))}
+      </DropdownMenu.Items>
+    </DropdownMenu>
+  );
+}
+
+const ROLES_DATA: Record<
+  RoleType,
+  { description: string; color: "red" | "amber" | "emerald" | "slate" }
+> = {
+  admin: {
+    description: "Admins can manage members, in addition to builders' rights.",
+    color: "red",
+  },
+  builder: {
+    description:
+      "Builders can create custom assistants and use advanced dev tools.",
+    color: "amber",
+  },
+  user: {
+    description:
+      "Members can use assistants provided by Dust as well as custom assistants created by their company.",
+    color: "emerald",
+  },
+  none: {
```

Comment:
So from what I understand `none` is not `revoked` -> `revoked` is `revoked` and that's the status we have in bd for users revoked, and `none` is when auth was done internally without a user. 
Maybe this has changed recently but I don't think we need this here, we could have ROLES_DATA being a record of ActiveRoleType only?

Reply:
AFAICT "revoked" in DB = "none" in the  membership type, this is how this page initially worked (e.g. see when revoking members, passes none).
I'd agree this probably needs a refactoring but IMHO outside the scope of the PR
I had tried to do only ActiveRoleType for ROLES_DATA, but didn't work because theoretically `member.workspace[0].role` can be none
Practically it's not possible in this page but typescript can't infer it
But you know i'll give it another shot, thanks for pushing

Reply:
Agreed that refactoring is out of scope of your PR but indeed might be worth flagging a refacto task somewhere, I might be wrong but this looks fishy to have revoked on db = none on front and none for users 😄

Reply:
Agreed, will try to tackle as a trail of the easier onboarding project